### PR TITLE
Interoperability with dpctl

### PR DESCRIPTION
**Pull request by michael-smirnov**
Fix of the **Data parallel control** module integration (formerly known as DPPY, DPPL). All the code that depends on dpctl is moved to a separate module to avoid unwanted dependency in the big daal4py.

### CODE REVIEW

**Review by PetrovKP — APPROVED** [review]
LGTM

File: setup.py [hunk 1]
```diff
@@ -43,6 +43,8 @@
 mpi_root = None if no_dist else os.environ['MPIROOT']
 dpcpp = True if 'DPCPPROOT' in os.environ else False
 dpcpp_root = None if not dpcpp else os.environ['DPCPPROOT']
+dpctl = True if dpcpp and 'DPCTLROOT' in os.environ else False
```
Comment (napetrov) [inline setup.py:46]: What if we will use dpctl from conda and not from dpctlroot?

**Comment by napetrov** [comment]
Overall looks ok - and we might merge in this configuration.  The only question that is not clear - how we will work in case of dpctl use from conda

File: conda-recipe/bld.bat [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+    set "CXX=dpcpp.exe"
+)
+
+rem if DAALROOT not exists then provide PREFIX
+rem otherwise use library directly from the path provided
+
+IF NOT DEFINED DAALROOT ( 
+    set DAALROOT=%PREFIX% 
+) ELSE (
+    call conda remove daal daal-include --force -y
```
Comment (napetrov) [inline conda-recipe/bld.bat:19]: should we also remove daal-devel?

**Review by napetrov — APPROVED** [review]

**Pull request merged**